mobile/envoy_build_config/BUILD

```diff
@@ -56,6 +52,14 @@ envoy_cc_library(
             "@envoy//source/common/quic:quic_transport_socket_factory_lib",
         ],
         "@envoy",
+    ) + envoy_select_envoy_mobile_stats_reporting(
+        [
+            "@envoy//source/extensions/stat_sinks/metrics_service:config",
+            "@envoy//source/extensions/stat_sinks/statsd:config",
+            "@envoy//source/extensions/clusters/logical_dns:logical_dns_cluster_lib",
```

Comment:
Don't we need this cluster for happy eyeballs?

Reply:
Happy eyeballs works with DFP, no?

Looks to me like we use logical DNS for direct responses (in swift tests) and for stats clusters (library/cc/engine_builder.cc LOGICAL_DNS)

Reply:
another reason to cut direct responses from our APIs. Hopefully we will have time to prioritize this on our end (Lyft) sooner than later.

Reply:
I'm sure you're right, but just to make sure I understand... The docs talk about LOGICAL_DNS, but I guess that's the DiscoveryType not the cluster type?
https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/connection_pooling#happy-eyeballs-support
And in one of the PRs which wires up Happy Eyeballs, we modifed logical_host.cc, but that's also not the same thing as the logical DNS *cluster*?
https://github.com/envoyproxy/envoy/pull/17371/files#diff-83c925d22248fe7ab6cad6424953960a158f879f8a1bfa45b510c64ccceb913b

So nothing above requires the logical_dns_cluster to work, right?

Reply:
yeah for DFP we configure the same `ALL` but in the cache config which goes in the DFP dns_cache_config, boostrap config, etc.